internal/storage/elasticsearch/config/config.go

```diff
@@ -641,29 +643,25 @@ func GetHTTPRoundTripper(ctx context.Context, c *Configuration, logger *zap.Logg
 		transport.TLSClientConfig = tlsConfig
 	}
 
-	// Wrap with authentication layer if configured.
-	var roundTripper http.RoundTripper = transport
+	// Initialize authentication methods.
+	var authMethods []auth.Method
 
+	// Bearer Token Authentication
 	if c.Authentication.BearerTokenAuthentication.HasValue() {
 		bearerAuth := c.Authentication.BearerTokenAuthentication.Get()
-		if bearerAuth.AllowFromContext || bearerAuth.FilePath != "" {
-			token := ""
-			if bearerAuth.FilePath != "" {
-				if bearerAuth.AllowFromContext {
-					logger.Warn("Token file and token propagation are both enabled, token from file won't be used")
-				}
-				tokenFromFile, err := loadTokenFromFile(bearerAuth.FilePath)
-				if err != nil {
-					return nil, err
-				}
-				token = tokenFromFile
-			}
+		ba, err := initBearerAuth(bearerAuth, logger)
+		if err != nil {
```

Comment:
It looks like there is a condition where an empty `Method{}` is returned. Maybe return pointers instead to indicate "not applicable" with nil? The slice can still be by-value.